app/controllers/api/v2/template_controller.rb

```diff
@@ -22,6 +23,28 @@ def import
         }).import!
         render :json => { :message => results }
       end
+
+      api :POST, "/template/export", N_("Initiate Export")
+      param :verbose, :bool, :required => false, :desc => N_("Set verbosity of export")
+      param :repo, String, :required => false, :desc => N_("Override the default repo from settings")
+      param :branch, String, :required => false, :desc => N_("Branch in Git repo.")
+      param :filter, String, :required => false, :desc => N_("Export templates with names matching this regex (case-insensitive; snippets are not filtered).")
+      param :negate, :bool, :required => false, :desc => N_("Negate the prefix (for purging).")
+      param :export_mode, String, :required => false, :desc => N_("Specify how to handle metadata ('refresh', 'remove', 'keep')")
+      def export
+        ForemanTemplates::TemplateExporter.new({
+          verbose:              params['verbose'],
+          repo:                 params['repo'],
+          branch:               params['branch'],
+          filter:               params['filter'],
+          negate:               params['negate'],
+          metadata_export_mode: params['export_mode']
```

Comment:
is there a reason why the param is named differently than the setting name? also I think `dirname` is missing here and in the docs above